client/dive-common/components/ControlsContainer.vue

```diff
@@ -115,6 +115,42 @@ export default defineComponent({
             >
               {{ imageData[frame].filename }}
             </span>
+            <span v-else-if="datasetType === 'video'">
+              <span class="mr-2">
+                <v-menu
+                  top
+                  offset-y
+                  nudge-left="3"
+                  open-on-hover
+                  :close-on-content-click="false"
+                  close-delay="500"
+                  rounded="pill"
+                >
+                  <template v-slot:activator="{ on }">
+                    <v-icon
+                      @click="(!volume && setVolume(1)) || (volume && setVolume(0))"
+                      v-on="on"
+                    > {{ volume === 0 ? 'mdi-volume-off' :'mdi-volume-medium' }}
+                    </v-icon>
+                  </template>
+                  <v-card style="overflow:hidden; width:30px">
+                    <v-slider
+                      :value="volume"
+                      min="0"
+                      max="1.0"
+                      step="0.05"
+                      vertical
+                      @change="setVolume"
+                    />
+                  </v-card>
+                </v-menu>
+              </span>
+              <span
+                class="text-middle"
+              >
+                {{ time }}
+              </span>
```

Comment:
I tend to avoid using variables that could cause a high-framerate re-render of an entire component.  

I've gotten into places where this will cause a large component to redraw even though only this span changes.  Certain vuetify components (like v-menu) have abnormally high rendering cost.

You can mitigate this by sticking this span into its own SFC that injects the time variable alone. You don't notice the problem on low-framerate settings (video is usually 5hz still) but when you crank it to 30 or 60, a 20% reduction in render loop cost starts to make a difference for commodity hardware.

In this component, there are already several performance issues, so this one isn't going be the last straw or anything.  

![Screenshot from 2021-04-19 12-37-11](https://user-images.githubusercontent.com/4214172/115272171-1a6d9200-a10c-11eb-8158-70f5d38a6c89.png)
![Screenshot from 2021-04-19 11-38-01](https://user-images.githubusercontent.com/4214172/115272173-1a6d9200-a10c-11eb-8b8f-0c65dd7300ed.png)

Reply:
Guessing this applies as well for the image-sequence if it is at a high enough frame rate?  Thoughts on including one SFC for both the ImageAnnotator filename (for the image in the sequence) and the Video currentTime/duration?  Just one component with prop to determine what displays and it handles the filename/currentTime/duration injection from useMediaController.

Reply:
That would be fine.  Also, no need to do it now, just wanted to mention it.